### PR TITLE
updates to intersection identification

### PR DESCRIPTION
**Pull request by scottgratton**
## Purpose
Causing this failure in `promote`
https://app.bugsnag.com/adwerx/promote/errors/60141604e804d60018ebbe3e

The address `Lot 55w & 56w Long Rifle Rd, Sevierville, TN 37862, US` was getting identified as an intersection, but had no `street` in the parsed output:
```
{
          "street" => nil,
     "street_type" => nil,
          "suffix" => nil,
          "prefix" => nil,
            "city" => "Sevierville",
           "state" => "TN",
     "postal_code" => "37862",
         "street2" => "56w Long Rifle",
    "street_type2" => "Rd",
         "suffix2" => nil,
         "prefix2" => nil
}
```
Because it _did_ have a `street2`, though, it was getting identified as an intersection, but when asking for `line1` the code there was trying to do essentially `'' += nil` since `street` was `nil`.

# Approach
For that reason, the check if a parsed address is an intersection will look at the presence of both `street` _and_ `street2`. I added a test case and was able to run the tests with `rake test` to catch a few other finicky things. The address itself won't parse perfectly-- not finding the street name, but unless I want to get into the regex there (I don't), I think just ensuring this doesn't error is good enough for now.

### CODE REVIEW

File: lib/street_address.rb [hunk 1]
```diff
@@ -852,7 +852,7 @@ def state_name
       end
 
       def intersection?
-        !street2.nil?
+        !!street && !!street2
```
Comment (scottgratton) [inline lib/street_address.rb:855]: Tricky rails stuff, but if I just did `street && street2`, the output value of `intersection?` was actually a string value (albeit, truthy), and that broke the assertions of some tests. For that reason, went with `!!`.

Comment (jbielick) [reply lib/street_address.rb:855]: totally up to you, but you could update those tests to `assert addr.intersection?` and `refute addr.intersection?` where it is and is not an intersection, respectively.

File: lib/street_address.rb [hunk 2]
```diff
@@ -867,7 +867,7 @@ def line1(s = "")
           s += " " + street_type2 unless street_type2.nil?
           s += " " + suffix2 unless suffix2.nil?
         else
-          s += number
+          s += number unless number.nil?
```
Comment (jbielick) [inline lib/street_address.rb:870]: is this the next thing that fails? this makes sense given there's no case for "is a street (no number)", but one side effect is the next few portions may have an extraneous space I guess. i.e. the next lines prepend a space assuming there was always a number starting the string.

File: test/test_street_address.rb [hunk 3]
```diff
@@ -139,6 +140,17 @@ def test_parse
     assert_equal addr.state, 'WI'
     assert_equal addr.postal_code, '53949'
 
+    # not a perfect parse, but this asserts not getting wrong data or an error
+    addr = StreetAddress::US.parse(@addr9)
+    assert_equal addr.line1, ''
+    assert_equal addr.number, nil
+    assert_equal addr.street, nil
```
Comment (jbielick) [inline test/test_street_address.rb:147]: I don't know if anyone wants to battle that regex, but I guess what would be nice here is if the street in the address string actually appeared here.

**Review by jbielick — APPROVED** [review]
👍🏻

File: test/test_street_address.rb (hunk 3 above)
Comment (scottgratton) [reply test/test_street_address.rb:147]: I agree for sure-- that's why I felt compelled to leave the comment above the block here. I think this is enough of an edge case that I'm okay just getting the city/state/zip back, unless you feel like that violates the expectation from the return value of this gem.

File: lib/street_address.rb (hunk 1 above)
Comment (scottgratton) [reply lib/street_address.rb:855]: Out of principle, I feel like a `?` method should return a boolean and that was the expectation before, so I'd personally go with a statement that holds to that rather than change the tests. 🤷‍♂️

**Review by walt1020 — APPROVED** [review]

File: lib/street_address.rb (hunk 2 above)
Comment (scottgratton) [reply lib/street_address.rb:870]: good catch-- 🤔

Comment (scottgratton) [reply lib/street_address.rb:870]: how do you feel about just adding a `squish` somewhere... maybe just on the return value at the end 🤔

Comment (scottgratton) [reply lib/street_address.rb:870]: ooh-- or `strip` I guess is the right one

File: lib/street_address.rb [hunk 4]
```diff
@@ -880,7 +880,7 @@ def line1(s = "")
           s += " " + suffix unless suffix.nil?
         end
 
-        return s
+        s.strip
```
Comment (scottgratton) [inline lib/street_address.rb:883]: @jbielick ☝️  force pushed this-- that should deal with the leading space in those edge cases where there's no number
